bgpd/bgp_route.c

```diff
@@ -6542,16 +6542,25 @@ static void bgp_aggregate_install(
 
 		if (!attr) {
 			bgp_aggregate_delete(bgp, p, afi, safi, aggregate);
+			if (BGP_DEBUG(update_groups, UPDATE_GROUPS))
+				zlog_debug("%s(%pFX): null attribute", __func__,
+					   p);
 			return;
 		}
 
-		new = info_make(ZEBRA_ROUTE_BGP, BGP_ROUTE_AGGREGATE, 0,
-				bgp->peer_self, attr, dest);
+		/* Do not install the aggregate route if BGP is in the
+		 * process of termination.
+		 */
+		if (bgp->peer_self
+		    && !CHECK_FLAG(bgp->flags, BGP_FLAG_DELETE_IN_PROGRESS)) {
```

Comment:
this test needs to be higher up in the call path.  We already have one place in the aggregate code where we look for the delete in progress.  *if* we are going down we do not want to do the work that figures out that we should add a new aggregage in.  Better to abort early instead of here.  From the code commit I could not see the actual backtrace.  Can you share that?

Reply:
bgp_aggregate_route -> bgp_aggregate_install
bgp_add_route_to_aggregate -> bgp_aggregate_install
bgp_remove_route_from_aggregate -> bgp_aggregate_install
bgp_aggregate_unset -> bgp_aggregate_install

This path `bgp_aggregate_route -> bgp_aggregate_install` is already covered (has the same check). We should do the same with others like @donaldsharp said to return as early as possible in the code path.

Reply:
I have done the suggested changes.

Reply:
@donaldsharp : @ton31337 has already approved the changes. Please merge the changes. I am closing the conversation.